sgeop/simplify.py

```diff
@@ -198,162 +248,175 @@ def simplify_singletons(
+        are no longer than this value. Must be greater than 0.
+        Used in multiple internal geometric operations.
+    min_dangle_length : float | int = 20
+        The threshold for determining if linestrings are dangling slivers to be
+        removed or not.
+    clip_limit : float | int = 2
+        Following generation of the Voronoi linework, we clip to fit inside the
+        polygon. To ensure we get a space to make proper topological connections
+        from the linework to the actual points on the edge of the polygon, we clip
+        using a polygon with a negative buffer of ``clip_limit`` or the radius of
+        maximum inscribed circle, whichever is smaller.
+    simplification_factor : float | int = 2
+        The factor by which singles, pairs, and clusters are simplified. The
+        ``max_segment_length`` is multiplied by this factor to get the
+        simplification epsilon.
+    consolidation_tolerance : float | int = 10
+        Tolerance passed to node consolidation when generating Voronoi skeletons.
 
     Returns
     -------
-
+    geopandas.GeoDataFrame
+        The road network line data following the pairs procedure.
     """
 
-    # Get nodes from the network.
-    nodes = momepy.nx_to_gdf(momepy.node_degree(momepy.gdf_to_nx(roads)), lines=False)
-
-    # Link nodes to artifacts
-    node_idx, artifact_idx = artifacts.sindex.query(
-        nodes.buffer(0.1), predicate="intersects"
-    )
-    intersects = sparse.coo_array(
-        ([True] * len(node_idx), (node_idx, artifact_idx)),
-        shape=(len(nodes), len(artifacts)),
-        dtype=np.bool_,
-    )
-
-    # Compute number of nodes per artifact
-    artifacts["node_count"] = intersects.sum(axis=0)
+    # Extract network nodes and relate to artifacts
+    nodes, artifacts = _link_nodes_artifacts("pairs", roads, artifacts, None)
 
     # Compute number of stroke groups per artifact
     roads, _ = continuity(roads)
-    strokes, c_, e_, s_ = get_stroke_info(artifacts, roads)
-
-    artifacts["stroke_count"] = strokes
-    artifacts["C"] = c_
-    artifacts["E"] = e_
-    artifacts["S"] = s_
+    artifacts = _classify_strokes(artifacts, roads)
 
-    # Filter artifacts caused by non-planar intersections.
-    artifacts["non_planar"] = artifacts["stroke_count"] > artifacts["node_count"]
-    a_idx, _ = roads.sindex.query(artifacts.geometry.boundary, predicate="overlaps")
-    artifacts.loc[artifacts.index[np.unique(a_idx)], "non_planar"] = True
+    # Filter artifacts caused by non-planar intersections
+    artifacts = _identify_non_planar(artifacts, roads)
 
-    artifacts["non_planar_cluster"] = artifacts.apply(
-        lambda x: sum(artifacts.loc[artifacts["comp"] == x.comp]["non_planar"]), axis=1
-    )
+    # Identify non-planar clusters
+    _id_np = lambda x: sum(artifacts.loc[artifacts["comp"] == x.comp]["non_planar"])  # noqa: E731
+    artifacts["non_planar_cluster"] = artifacts.apply(_id_np, axis=1)
```

Comment:
Sometimes I wonder if refactors of this sort are worth it. I don't consider the original less readable (maybe more), and it does not require linting exception.

Reply:
While I think this is more readable, I completely understand your view. I have no problem reverting.